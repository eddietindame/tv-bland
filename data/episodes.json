[
  {
    "id": 2042439,
    "url": "https://www.tvmaze.com/episodes/2042439/elliott-from-earth-1x05-idiosyncratic-induction",
    "name": "Idiosyncratic Induction",
    "season": 1,
    "number": 5,
    "type": "regular",
    "airdate": "2021-03-30",
    "airtime": "09:00",
    "airstamp": "2021-03-30T13:00:00+00:00",
    "runtime": 15,
    "image": null,
    "summary": "<p>On Elliott's first day at the Hive, he meets the most unusual characters on his adventure so far.</p>",
    "show": {
      "id": 52246,
      "url": "https://www.tvmaze.com/shows/52246/elliott-from-earth",
      "name": "Elliott from Earth",
      "type": "Animation",
      "language": "English",
      "genres": ["Comedy", "Science-Fiction"],
      "status": "Running",
      "runtime": 15,
      "premiered": "2021-03-29",
      "officialSite": "https://elliottfromearth.cartoonnetwork.co.uk/",
      "schedule": {
        "time": "",
        "days": []
      },
      "rating": {
        "average": null
      },
      "weight": 99,
      "network": {
        "id": 11,
        "name": "Cartoon Network",
        "country": {
          "name": "United States",
          "code": "US",
          "timezone": "America/New_York"
        }
      },
      "webChannel": null,
      "dvdCountry": null,
      "externals": {
        "tvrage": null,
        "thetvdb": null,
        "imdb": null
      },
      "image": {
        "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/298/747331.jpg",
        "original": "https://static.tvmaze.com/uploads/images/original_untouched/298/747331.jpg"
      },
      "summary": "<p><b>Elliott from Earth</b> is a comedy animated series about a young boy who finds himself living on a gigantic spaceship, with all sorts of aliens from around the universe. It's a show about finding your place in the world, even if it's in outer space. Oh, and his best friend is a dinosaur.</p>",
      "updated": 1617081181,
      "_links": {
        "self": {
          "href": "https://api.tvmaze.com/shows/52246"
        },
        "previousepisode": {
          "href": "https://api.tvmaze.com/episodes/2042440"
        },
        "nextepisode": {
          "href": "https://api.tvmaze.com/episodes/2042441"
        }
      },

      "_embedded": {
        "cast": [
          {
            "person": {
              "id": 172,
              "url": "https://www.tvmaze.com/people/172/laura-vandervoort",
              "name": "Laura Vandervoort",
              "country": {
                "name": "Canada",
                "code": "CA",
                "timezone": "America/Halifax"
              },
              "birthday": "1984-09-22",
              "deathday": null,
              "gender": "Female",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/0/1751.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/0/1751.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/people/172"
                }
              }
            },
            "character": {
              "id": 125,
              "url": "https://www.tvmaze.com/characters/125/bitten-elena-michaels",
              "name": "Elena Michaels",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/0/16.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/0/16.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/characters/125"
                }
              }
            },
            "self": false,
            "voice": false
          },
          {
            "person": {
              "id": 173,
              "url": "https://www.tvmaze.com/people/173/greyston-holt",
              "name": "Greyston Holt",
              "country": {
                "name": "Canada",
                "code": "CA",
                "timezone": "America/Halifax"
              },
              "birthday": "1985-09-30",
              "deathday": null,
              "gender": "Male",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/214/536036.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/214/536036.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/people/173"
                }
              }
            },
            "character": {
              "id": 126,
              "url": "https://www.tvmaze.com/characters/126/bitten-clayton-danvers",
              "name": "Clayton Danvers",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/8/22148.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/8/22148.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/characters/126"
                }
              }
            },
            "self": false,
            "voice": false
          },
          {
            "person": {
              "id": 174,
              "url": "https://www.tvmaze.com/people/174/greg-bryk",
              "name": "Greg Bryk",
              "country": {
                "name": "Canada",
                "code": "CA",
                "timezone": "America/Halifax"
              },
              "birthday": "1972-01-01",
              "deathday": null,
              "gender": "Male",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/36/90548.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/36/90548.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/people/174"
                }
              }
            },
            "character": {
              "id": 127,
              "url": "https://www.tvmaze.com/characters/127/bitten-jeremy-danvers",
              "name": "Jeremy Danvers",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/0/17.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/0/17.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/characters/127"
                }
              }
            },
            "self": false,
            "voice": false
          },
          {
            "person": {
              "id": 176,
              "url": "https://www.tvmaze.com/people/176/steve-lund",
              "name": "Steve Lund",
              "country": {
                "name": "Canada",
                "code": "CA",
                "timezone": "America/Halifax"
              },
              "birthday": "1989-01-09",
              "deathday": null,
              "gender": "Male",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/161/404443.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/161/404443.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/people/176"
                }
              }
            },
            "character": {
              "id": 129,
              "url": "https://www.tvmaze.com/characters/129/bitten-nick-sorrentino",
              "name": "Nick Sorrentino",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/0/19.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/0/19.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/characters/129"
                }
              }
            },
            "self": false,
            "voice": false
          },
          {
            "person": {
              "id": 192,
              "url": "https://www.tvmaze.com/people/192/michael-xavier",
              "name": "Michael Xavier",
              "country": {
                "name": "Canada",
                "code": "CA",
                "timezone": "America/Halifax"
              },
              "birthday": null,
              "deathday": null,
              "gender": "Male",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/160/401035.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/160/401035.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/people/192"
                }
              }
            },
            "character": {
              "id": 130,
              "url": "https://www.tvmaze.com/characters/130/bitten-logan-jonsen",
              "name": "Logan Jonsen",
              "image": {
                "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/8/22149.jpg",
                "original": "https://static.tvmaze.com/uploads/images/original_untouched/8/22149.jpg"
              },
              "_links": {
                "self": {
                  "href": "https://api.tvmaze.com/characters/130"
                }
              }
            },
            "self": false,
            "voice": false
          }
        ]
      }
    },
    "_links": {
      "self": {
        "href": "https://api.tvmaze.com/episodes/2042439"
      }
    }
  },
  {
    "id": 2054362,
    "url": "https://www.tvmaze.com/episodes/2054362/the-young-and-the-restless-2021-03-30-ep-12076",
    "name": "Ep. #12076",
    "season": 2021,
    "number": 55,
    "type": "regular",
    "airdate": "2021-03-30",
    "airtime": "12:30",
    "airstamp": "2021-03-30T16:30:00+00:00",
    "runtime": 60,
    "image": null,
    "summary": null,
    "show": {
      "id": 4344,
      "url": "https://www.tvmaze.com/shows/4344/the-young-and-the-restless",
      "name": "The Young and the Restless",
      "type": "Scripted",
      "language": "English",
      "genres": ["Drama", "Family", "Romance"],
      "status": "Running",
      "runtime": 60,
      "premiered": "1973-03-26",
      "officialSite": "http://www.cbs.com/shows/the_young_and_the_restless/",
      "schedule": {
        "time": "12:30",
        "days": ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
      },
      "rating": {
        "average": 6.2
      },
      "weight": 96,
      "network": {
        "id": 2,
        "name": "CBS",
        "country": {
          "name": "United States",
          "code": "US",
          "timezone": "America/New_York"
        }
      },
      "webChannel": null,
      "dvdCountry": null,
      "externals": {
        "tvrage": 6318,
        "thetvdb": 70328,
        "imdb": "tt0069658"
      },
      "image": {
        "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/233/583614.jpg",
        "original": "https://static.tvmaze.com/uploads/images/original_untouched/233/583614.jpg"
      },
      "summary": "<p><b>The Young and the Restless</b> revolves around the rivalries, romances, hopes and fears of the residents of the fictional Midwestern metropolis, Genoa City. The lives and loves of a wide variety of characters mingle through the generations, dominated by the Newman, Abbott, Chancellor, Baldwin and Winters families. ­ When The Young and the Restless premiered in 1973, it revolutionized the daytime drama. It continues to set the standard with strong characters, socially conscious storylines, romance and sensuality.</p>",
      "updated": 1616892558,
      "_links": {
        "self": {
          "href": "https://api.tvmaze.com/shows/4344"
        },
        "previousepisode": {
          "href": "https://api.tvmaze.com/episodes/2050236"
        },
        "nextepisode": {
          "href": "https://api.tvmaze.com/episodes/2054362"
        }
      }
    },
    "_links": {
      "self": {
        "href": "https://api.tvmaze.com/episodes/2054362"
      }
    }
  },
  {
    "id": 2055172,
    "url": "https://www.tvmaze.com/episodes/2055172/days-of-our-lives-2021-03-30-ep-14072",
    "name": "Ep. #14072",
    "season": 2021,
    "number": 62,
    "type": "regular",
    "airdate": "2021-03-30",
    "airtime": "13:00",
    "airstamp": "2021-03-30T17:00:00+00:00",
    "runtime": 60,
    "image": null,
    "summary": null,
    "show": {
      "id": 812,
      "url": "https://www.tvmaze.com/shows/812/days-of-our-lives",
      "name": "Days of Our Lives",
      "type": "Scripted",
      "language": "English",
      "genres": ["Drama", "Family", "Romance"],
      "status": "Running",
      "runtime": 60,
      "premiered": "1965-12-09",
      "officialSite": "http://www.nbc.com/days-of-our-lives",
      "schedule": {
        "time": "13:00",
        "days": ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
      },
      "rating": {
        "average": 6.4
      },
      "weight": 92,
      "network": {
        "id": 1,
        "name": "NBC",
        "country": {
          "name": "United States",
          "code": "US",
          "timezone": "America/New_York"
        }
      },
      "webChannel": null,
      "dvdCountry": null,
      "externals": {
        "tvrage": 3256,
        "thetvdb": 70366,
        "imdb": "tt0058796"
      },
      "image": {
        "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/87/218576.jpg",
        "original": "https://static.tvmaze.com/uploads/images/original_untouched/87/218576.jpg"
      },
      "summary": "<p><b>Days of our Lives</b> is set in the fictitious Midwestern town of Salem. The core families are the Bradys, the Hortons and the DiMeras, and the multi-layered storylines involve elements of romance, adventure, mystery, comedy and drama.</p>",
      "updated": 1617026539,
      "_links": {
        "self": {
          "href": "https://api.tvmaze.com/shows/812"
        },
        "previousepisode": {
          "href": "https://api.tvmaze.com/episodes/2055171"
        },
        "nextepisode": {
          "href": "https://api.tvmaze.com/episodes/2055172"
        }
      }
    },
    "_links": {
      "self": {
        "href": "https://api.tvmaze.com/episodes/2055172"
      }
    }
  }
]
