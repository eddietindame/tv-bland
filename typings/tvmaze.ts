export type ScheduleResponse = Episode[]

export interface Episode {
    id: number
    url: string // Url
    name: string
    season: number
    number: number
    type: string // "regular"
    airdate: string // "2021-03-30"
    airtime: string // 09:00
    airstamp: string // 2021-03-30T13:00:00+00:00 Date
    runtime: number
    image: Image | null
    summary: string | null // HTML formatted
    show: Show
    _links: Links
}

export interface Show {
    id: number
    url: string // Url
    name: string
    type: string
    language: string
    genres: string[] // "Comedy", "Science-Fiction"
    status: string // "Running"
    runtime: number
    premiered: string // 2021-03-29
    officialSite: string // Url
    schedule: ShowSchedule
    rating: ShowRating
    weight: number
    network: ShowNetwork
    webChannel: any // null
    dvdCountry: any // null
    externals: ShowExternals
    image: Image
    summary: string | null // HTML formatted
    updated: number // Timestamp
    _links: Links
}

export interface Image {
    medium: string // Url
    original: string // Url
}

export interface Links {
    self: { href: string }
    previousepisode?: { href: string }
    nextepisode?: { href: string }
}

export interface ShowSchedule {
    time: string // "13:30" | ""
    days: string[]
}

export interface ShowRating {
    average: number | null
}

export interface ShowNetwork {
    id: number
    name: string
    country: {
        name: string
        code: string // "US"
        timezone: string // "America/New_York"
    }
}

export interface ShowExternals {
    tvrage: number | null
    thetvdb: number | null
    imdb: string | null
}
