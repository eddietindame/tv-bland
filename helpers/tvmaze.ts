import axios from 'axios'
import { Schedule, ShowWithCast, ShowSchedule } from '@/typings/tvmaze'

export const tvmaze = axios.create({
    baseURL: 'https://api.tvmaze.com'
})

export const getSchedule = () =>
    tvmaze({ url: '/schedule' }).then(({ data }: { data: Schedule }) => data)

export const getShowWithCastById = (id: number) =>
    tvmaze({
        url: `/shows/${id}`,
        params: { embed: 'cast' }
    }).then(({ data }: { data: ShowWithCast }) => data)

/**
 * Formats show schedule into displayable string
 * @param schedule - ShowSchedule type (object)
 * @returns string
 */
export const formatSchedule = (schedule: ShowSchedule) => {
    let days

    if (schedule.days.length === 7) days = 'Every day'
    else if (
        schedule.days.length === 5 &&
        schedule.days.includes('Monday') &&
        schedule.days.includes('Tuesday') &&
        schedule.days.includes('Wednesday') &&
        schedule.days.includes('Thursday') &&
        schedule.days.includes('Friday')
    )
        days = 'Weekdays'
    else {
        if (schedule.days.length > 1) {
            const lastDay = schedule.days.pop()
            days =
                schedule.days.map(day => day + 's').join(', ') +
                ` & ${lastDay}s`
        } else days = schedule.days[0] + 's'
    }

    const time = schedule.time ? ` at ${schedule.time}` : ''

    return days + time
}
