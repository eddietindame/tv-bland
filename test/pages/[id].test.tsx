import React from 'react'

import { render } from '@/testUtils'
import { ShowWithCast } from '@/typings/tvmaze'
import episodes from '@/data/episodes.json'
import TVShowPage from '@/pages/shows/[id]'

describe('TVShowPage page', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(
            <TVShowPage show={episodes[0].show as ShowWithCast} />
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
