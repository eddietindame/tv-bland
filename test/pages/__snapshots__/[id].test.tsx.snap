// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TVShowPage page matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="show-layout-top"
  >
    <div
      class="show-layout-top__inner"
    >
      <h1
        class="show-layout-top__heading"
      >
        <a
          class="show-layout-top__heading__anchor"
          href="/"
        >
          TV Bland
        </a>
      </h1>
      <div
        class="grid-row"
      >
        <div
          class="show-layout-top__image-wrap"
        >
          <img
            alt="Elliott from Earth"
            src="https://static.tvmaze.com/uploads/images/original_untouched/298/747331.jpg"
          />
        </div>
        <div
          class="show-layout-top__info-wrap"
        >
          <div
            class="show-layout-top__rating"
          >
            Rating unavailable
          </div>
          <h2
            class="show-layout-top__show-title"
          >
            Elliott from Earth
          </h2>
          <div
            class="show-layout-top__summary"
          >
            <p>
              <b>
                Elliott from Earth
              </b>
               is a comedy animated series about a young boy who finds himself living on a gigantic spaceship, with all sorts of aliens from around the universe. It's a show about finding your place in the world, even if it's in outer space. Oh, and his best friend is a dinosaur.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    class="show-layout-bottom"
  >
    <div
      class="show-layout-top__inner"
    >
      <div
        class="grid-row"
      >
        <div
          class="show-layout-bottom__col"
        >
          <h3>
            Show info
          </h3>
          <ul>
            <li>
              <div>
                <div>
                  Streamed on
                </div>
                <div>
                  Cartoon Network
                </div>
              </div>
            </li>
            <li>
              <div>
                <div>
                  Schedule
                </div>
                <div>
                  Not available
                </div>
              </div>
            </li>
            <li>
              <div>
                <div>
                  Status
                </div>
                <div>
                  Running
                </div>
              </div>
            </li>
            <li>
              <div>
                <div>
                  Genres
                </div>
                <div>
                  Comedy, Science-Fiction
                </div>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="show-layout-bottom__col"
        >
          <h3>
            Starring
          </h3>
          <ul>
            <li>
              <div>
                <img
                  alt="Laura Vandervoort"
                  src="https://static.tvmaze.com/uploads/images/medium_portrait/0/1751.jpg"
                />
              </div>
              <div>
                <div>
                  Laura Vandervoort
                </div>
                <div>
                  Elena Michaels
                </div>
              </div>
            </li>
            <li>
              <div>
                <img
                  alt="Greyston Holt"
                  src="https://static.tvmaze.com/uploads/images/medium_portrait/214/536036.jpg"
                />
              </div>
              <div>
                <div>
                  Greyston Holt
                </div>
                <div>
                  Clayton Danvers
                </div>
              </div>
            </li>
            <li>
              <div>
                <img
                  alt="Greg Bryk"
                  src="https://static.tvmaze.com/uploads/images/medium_portrait/36/90548.jpg"
                />
              </div>
              <div>
                <div>
                  Greg Bryk
                </div>
                <div>
                  Jeremy Danvers
                </div>
              </div>
            </li>
            <li>
              <div>
                <img
                  alt="Steve Lund"
                  src="https://static.tvmaze.com/uploads/images/medium_portrait/161/404443.jpg"
                />
              </div>
              <div>
                <div>
                  Steve Lund
                </div>
                <div>
                  Nick Sorrentino
                </div>
              </div>
            </li>
            <button>
              Show more
            </button>
          </ul>
        </div>
      </div>
    </div>
  </section>
  <footer>
    <div>
      Â© TV Bland 2021 
      <span>
        |
      </span>
       
      <a
        href="#"
      >
        Cookies
      </a>
       
      <span>
        |
      </span>
       
      <a
        href="#"
      >
        Privacy
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
